parity/configuration.rs

```diff
@@ -241,7 +241,7 @@ impl Configuration {
 	pub fn network_settings(&self) -> NetworkSettings {
 		NetworkSettings {
 			name: self.args.flag_identity.clone(),
-			chain: self.args.flag_chain.clone(),
```

Comment:
why is this even needed? i means that `flag_testnet` and `flag_chain` are interpreted in two different positions, leading to this bug. args should be interpreted in only one position and network shouldn't need to care about what chain it is.